PyInstaller/compat.py

```diff
@@ -581,26 +533,42 @@ def expand_path(path):
     """
     return os.path.expandvars(os.path.expanduser(path))
 
+
 # Site-packages functions - use native function if available.
-if hasattr(site, 'getsitepackages'):
-    getsitepackages = site.getsitepackages
+def getsitepackages(prefixes=None):
```

Comment:
Call this `def _getsitepackages(prefixes=None)`.

Reply:
See the *new* line 571 to see why J didn't.

Reply:
Hmmm, I still don't get it -- my suggestion is to avoid renaming an existing symbol, which seems a bit unnecessary.

Also, why the new implementation? Is this copied from/tested elsewhere? This seems like a big change.

Reply:
@bjones1 the new implementation is copied straight from `site`. I don't know why it wasn't in the first place.

> Hmmm, I still don't get it -- my suggestion is to avoid renaming an existing symbol, which seems a bit unnecessary.

I'm using `getsitepackages = getattr(site, 'getsitepackages', getsitepackages)` below.

This means we don't have to delete `_getsitepackages` below like before. Yes I could add an underscore and override, but I don't see the point of extra code.

Reply:
OK, thanks. Would you mind adding a comment to state the origin of this code, and the version of Python it came from? That will be helpful if we need to update it for future versions of Python.

I see re: delete vs. hide. Sounds good. Thanks again for working on this!

Reply:
@bjones1 en route.